### PR TITLE
fix #43 to compile with c++17

### LINKED ISSUE #43: fcalcclusterer fails to compile with stdc++-17
fcalcclusterer fails to compile when set to stdc++-17 because std::random_shuffle has been removed

/tmp/user/spack-stage/spack-stage-fcalclusterer-1.0.1-6uszwdiiq25ca4yyeofqasn6mufg6tmb/spack-src/source/BeamCalReco/src/BeamCalBkgPregen.cpp:70:8: error: no member named 'random_shuffle' in namespace 'std' [clang-diagnostic-error]
  std::random_shuffle(bg_files.begin(), bg_files.end());
       ^
1 error generated.
Error while processing /tmp/user/spack-stage/spack-stage-fcalclusterer-1.0.1-6uszwdiiq25ca4yyeofqasn6mufg6tmb/spack-src/source/BeamCalReco/src/BeamCalBkgPregen.cpp.
Found compiler error(s).
:

### PR DESCRIPTION
**Pull request by joequant**

### CODE REVIEW

**Comment by vvolkl** [comment]
Thanks @joequant! Can you submit this patch also to the upstream repo: https://github.com/FCALSW/FCalClusterer/ ?

File: packages/fcalclusterer/package.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ class Fcalclusterer(CMakePackage):
 
     # CMAKE_INSTALL_PREFIX is overwritten by the package
     patch("install.patch", when="@:1.0.1")
+    patch("random-shuffle-c17.patch")
```
Comment (vvolkl) [inline packages/fcalclusterer/package.py:32]: From the issue you opened, I gather this has been fixed in master. In that case I'd suggest to do:
```suggestion
    patch("random-shuffle-c17.patch", when="@:1.0.1")
```

**Review by vvolkl — CHANGES_REQUESTED** [review]

**Comment by andresailer** [comment]
Superseded by #53 ?

**Comment by vvolkl** [comment]
#53 fixes the issue in `1.0.2`, but this will still be good to merge for `1.0.1`

**Pull request merged**